### PR TITLE
Revolution Improvements

### PR DESCRIPTION
**Pull request by Geevies**
* Added a new category to the traitor uplink, which is only available to revolutionaries.
* Head Revolutionaries now spawn with an uplink that has 50 TC, they also get an alert telling them to not abuse it.
* Revamped the drill dropper into a generic orbital dropper, capable of dropping ANY sort of item you wish it to.
* Added a Mech orbital dropper. Added an Armory orbital dropper.
* Made loyalty implants display their text correctly when they implant someone.
* Added aggression implants, which makes whoever it is implanted into extremely aggressive. Quite useful for starting a revolution.
* Added some description_antag to mercenary cipherkeys, informing the reader that the mercenary radio key is :t.
* Added a gear crate (not for mercs) which is just six syndicate softsuits. Useful in a pinch.

image
image
image

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by Geevies** [comment]
Forum Thread: https://forums.aurorastation.org/topic/13533-feedbackhype-revolution-improvements/

File: code/datums/trading/ai.dm [hunk 1]
```diff
@@ -50,7 +50,7 @@ They sell generic supplies and ask for generic supplies.
 		/obj/item/device/contract_uplink            = TRADER_BLACKLIST, // Traitor stuff
 		/obj/item/device/uplink                     = TRADER_BLACKLIST_ALL, // Traitor stuff
 		/obj/item/device/announcer                  = TRADER_BLACKLIST, // Rev item
-		/obj/item/device/ninja_uplink               = TRADER_BLACKLIST, // Ninja item
+		/obj/item/device/special_uplink/ninja       = TRADER_BLACKLIST, // Ninja item
```
Comment (Arrow768) [inline code/datums/trading/ai.dm]: Blacklist the rev uplink aswell or all /special_uplink

File: code/datums/uplink/revolution.dm [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+	item_cost = 10
+	path = /obj/item/storage/box/aggression
+
+/datum/uplink_item/item/revolution/encryption_keys
+	name = "Box of Encryption Keys"
+	desc = "A box of encryption keys that gives the user a safe channel to chatter in, while also allowing the user to intercept all departmental comms. Access safe channel with :t."
+	item_cost = 15
+	path = /obj/item/storage/box/encryption_key
+
+/datum/uplink_item/item/revolution/encryption_keys
```
Comment (Arrow768) [inline code/datums/uplink/revolution.dm]: Duplicate path

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 3]
```diff
@@ -778,4 +778,17 @@
...
+	starts_with = list(/obj/item/pen/fountain = 7)
+
+/obj/item/storage/box/aggression
+	name = "box"
+	desc = "It's just an ordinary box."
+	description_antag = "This box contains various implants that will make their owners increasingly aggressive."
+	max_storage_space = 12
+	starts_with = list(/obj/item/implantcase/aggression = 6, /obj/item/implanter = 1, /obj/item/implantpad = 1)
+
+/obj/item/storage/box/encryption_key
```
Comment (Arrow768) [inline code/game/objects/items/weapons/storage/boxes.dm]: It should grant a extra channel, but it should not allow to listen into other channels as that would be a rather unlevel playing field.

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 4]
```diff
@@ -778,4 +778,17 @@
...
+	name = "box"
+	desc = "It's just an ordinary box."
```
Comment (Arrow768) [inline code/game/objects/items/weapons/storage/boxes.dm]: Remove the desc here, so it inherits the default desc

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 5]
```diff
@@ -778,4 +778,17 @@
 
 /obj/item/storage/box/fountainpens
 	name = "box of fountain pens"
-	starts_with = list(/obj/item/pen/fountain = 7)
\ No newline at end of file
+	starts_with = list(/obj/item/pen/fountain = 7)
+
+/obj/item/storage/box/aggression
+	name = "box"
+	desc = "It's just an ordinary box."
```
Comment (Arrow768) [inline code/game/objects/items/weapons/storage/boxes.dm]: Remove the desc here, so it inherits the default desc

File: code/game/objects/items/weapons/implants/implant.dm [hunk 6]
```diff
@@ -576,3 +572,51 @@ the implant may become unstable and either pre-maturely inject the subject or si
...
+	<b>Implant Specifications:</b><BR>
+	<b>Name:</b> Aggression Implant<BR>
+	<b>Life:</b> N/A.<BR>
+	<b>Important Notes:</b> Users injected with this device get increasingly angry to a breaking point. Users tend to expire before the implant does.<BR>
+	<HR>
+	<b>Implant Details:</b><BR>
+	<b>Function:</b> Contains a small pod of nanobots that manipulate the host's mental functions.<BR>
+	<b>Integrity:</b> Implant will last so long as the nanobots are inside the bloodstream."}
+
+/obj/item/implant/aggression/implanted(mob/M)
```
Comment (Arrow768) [inline code/game/objects/items/weapons/implants/implant.dm]: You are not accounting for loyalty implanted personnel here.

File: code/game/objects/items/devices/uplink.dm [hunk 7]
```diff
@@ -387,21 +387,33 @@ A list of items and costs is stored under the datum of every game mode, alongsid
 	discord_bot.send_to_cciaa("Announcer - Fake announcement:`[title]` - `[message]`, sent by [user]!")
 	qdel(src)
 
-//ninja
-/obj/item/device/ninja_uplink
-	name = "infiltrator uplink"
-	desc = "A small device used for access to a restricted cache of specialized items."
+/obj/item/device/special_uplink
+	name = "special uplink"
+	desc = "A small device used to access a restricted cache of specialized items."
```
Comment (Arrow768) [inline code/game/objects/items/devices/uplink.dm]: Should probs rename that desc to something that describes only it´s external appearence and put the rest in antag_desc

File: code/datums/uplink/revolution.dm [hunk 8]
```diff
@@ -0,0 +1,32 @@
+/datum/uplink_item/item/revolution
+	category = /datum/uplink_category/revolution
+
+/datum/uplink_item/item/revolution/armory
+	name = "Armory Dropper"
+	desc = "A device that can be used to drop in an armory-worth of guns. Can only be used outside station areas, unless emagged, which is hazardous."
+	item_cost = 20
```
Comment (Arrow768) [inline code/datums/uplink/revolution.dm]: The price here needs to be upped severely to be in line with the rest of the syndie items.

File: code/datums/uplink/revolution.dm [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+	path = /obj/item/device/orbital_dropper/armory/syndicate
+
+/datum/uplink_item/item/revolution/mecha
+	name = "Mecha Dropper"
+	desc = "A device that can be used to drop in a combat exosuit. Can only be used outside station areas, unless emagged, which is hazardous."
+	item_cost = 25
```
Comment (Arrow768) [inline code/datums/uplink/revolution.dm]: The price needs to be increased, so its in line with the rest of the syndie items.

File: code/game/antagonist/station/revolutionary.dm [hunk 10]
```diff
@@ -7,7 +7,7 @@ var/datum/antagonist/revolutionary/revs
 	bantype = "revolutionary"
 	feedback_tag = "rev_objective"
 	antag_indicator = "rev_head"
-	welcome_text = "Down with the capitalists! Down with the Bourgeoise!"
+	welcome_text = "You are a Head Revolutionary! It's your job to whip the crew into a frenzy against Command and Central! <b>Use the uplink disguised as a station-bounced radio in your backpack to help start the revolution!</b>"
```
Comment (Arrow768) [inline code/game/antagonist/station/revolutionary.dm]: I disagree with the wording of that.
The purpose of the headrev isnt to start a deathmatch, it is to work with the head loyalists to provide a interesting story.

**Comment by Kaedwuff** [comment]
If the loyalty implant blocks the aggression implant, shouldn't the reverse be true too?  Otherwise loyalty implants, the tool of loyalists, is stronger than the rev implant.

File: code/game/objects/items/devices/drop_targeter/_droptargeter.dm [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+		return
+
+	var/turf/targloc = get_turf(target)
+	if(!emagged)
+		for(var/turf/t in block(locate(targloc.x+2,targloc.y+2,targloc.z), locate(targloc.x-2,targloc.y-2,targloc.z)))
+			if (!istype(t.loc, /area/mine))
+				to_chat(user, span("warning", "You can't do this so close to the station, point the laser further into the mine!"))
+				return
+			if (!isfloor(targloc))
+				to_chat(user, span("warning", "You cannot request a drill on unstable flooring!"))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/devices/drop_targeter/_droptargeter.dm]: This still says drill.

File: code/game/objects/items/devices/drop_targeter/_droptargeter.dm [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+
+/obj/item/device/orbital_dropper/proc/drill_drop(var/turf/target)
+	var/turf/east_turf
+	var/turf/south_east_turf
+	var/turf/south_turf
+	var/turf/south_west_turf
+	var/turf/west_turf
+	var/turf/north_west_turf
+	var/turf/north_turf
+	var/turf/north_east_turf
```
Comment (NonQueueingMatt) [inline code/game/objects/items/devices/drop_targeter/_droptargeter.dm]: It'd be far cleaner and better to assign all of these in the line where they're actually used.

File: code/game/objects/items/weapons/implants/implant.dm [hunk 13]
```diff
@@ -576,3 +572,57 @@ the implant may become unstable and either pre-maturely inject the subject or si
...
+	if(!istype(M, /mob/living/carbon/human))
+		return FALSE
+
+	var/mob/living/carbon/human/H = M
+
+	for(var/obj/item/implant/loyalty/I in H)
+		if(I.implanted)
+			to_chat(H, span("danger", "Rage surges through your body, but the nanobots from your loyalty implant stops it soon after it starts!"))
```
Comment (NonQueueingMatt) [inline code/game/objects/items/weapons/implants/implant.dm]:
```suggestion
			to_chat(H, span("danger", "Rage surges through your body, but the nanobots from your loyalty implant stop it soon after it starts!"))
```

File: code/game/objects/items/devices/drop_targeter/_droptargeter.dm (hunk 12 above)
Comment (NonQueueingMatt) [reply code/game/objects/items/devices/drop_targeter/_droptargeter.dm]: Also these should probably be a single assoc list of lists that gets iterated over with a loop.

File: code/game/objects/items/devices/drop_targeter/_droptargeter.dm [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+
+	flick_overlay(I, showto, 20) //2 seconds of the red dot appearing
+	icon_state = "drillpointer"
+
+/obj/item/device/orbital_dropper/proc/orbital_drop(var/turf/target)
+	var/dmm_suite/maploader = new
+	if(!maploader)
+		log_debug("Error, unable to load maploader in proc orbital_drop!")
+		return
+	maploader.load_map(file("maps/templates/" + template_name), target.x, target.y, target.z, no_changeturf = TRUE)
```
Comment (NonQueueingMatt) [inline code/game/objects/items/devices/drop_targeter/_droptargeter.dm]: This probably needs a log?

**Pull request merged**